esp32/modules/launcher.py

```diff
@@ -100,13 +98,42 @@ def perform_uninstall(ok):
 
 populate_it()
 
-ugfx.input_attach(ugfx.BTN_A, run_it)
-ugfx.input_attach(ugfx.BTN_SELECT, uninstall_it)
-
-ugfx.input_attach(ugfx.JOY_UP, lambda pushed: ugfx.flush() if pushed else 0)
-ugfx.input_attach(ugfx.JOY_DOWN, lambda pushed: ugfx.flush() if pushed else 0)
+def input_a(pressed):
+    pm.feed()
+    if pressed:
+        run_it()
+    
+def input_b(pressed):
+    pm.feed()
+    if pressed:
+        appglue.home()
 
-ugfx.input_attach(ugfx.BTN_B, lambda pushed: appglue.home() if pushed else 0)
-#ugfx.input_attach(ugfx.BTN_START, lambda pushed: appglue.home() if pushed else 0)
+def input_select(pressed):
+    pm.feed()
+    if pressed:
+        uninstall_it()
+    
+def input_other(pressed):
+    pm.feed()
+    if pressed:
+        ugfx.flush()
+
+ugfx.input_attach(ugfx.BTN_A, input_a)
+ugfx.input_attach(ugfx.BTN_B, input_b)
+ugfx.input_attach(ugfx.BTN_SELECT, input_select)
+ugfx.input_attach(ugfx.JOY_UP, input_other)
+ugfx.input_attach(ugfx.JOY_DOWN, input_other)
+ugfx.input_attach(ugfx.JOY_LEFT, input_other)
+ugfx.input_attach(ugfx.JOY_RIGHT, input_other)
+ugfx.input_attach(ugfx.BTN_START, input_other)
 
 ugfx.flush(ugfx.LUT_FULL)
+
+def pm_cb(dummy):
+    appglue.home()
+
+# Power management
+virtualtimers.activate(1000) # Start scheduler with 1 second ticks
+pm.set_timeout(5*60*1000) # Set timeout to 5 minutes
+pm.callback(pm_cb) # Go to splash instead of sleep
```

Comment:
would also be more readable as a lambda imho: `pm.callback(lambda: appglue.home())`